src/language_features/semantic_highlighting.rs

```diff
@@ -113,18 +111,11 @@ pub fn make_scope_map(ctx: &mut Context) -> std::vec::Vec<std::string::String> {
         .map(|(k, v)| (k.replace("_", "."), v))
         .collect();
 
-    let scopes = ctx
-        .capabilities
+    ctx.capabilities
         .as_ref()
         .and_then(|x| x.semantic_highlighting.as_ref())
-        .and_then(|x| x.scopes.as_ref());
-
-    if scopes.is_none() {
-        return Vec::new();
-    }
-    let scopes = scopes.unwrap();
-
-    map_scopes_to_faces(scopes, faces)
+        .and_then(|x| x.scopes.as_ref())
+        .map_or(Vec::new(), |v| map_scopes_to_faces(v, faces))
```

Comment:
If `Vec::new()` was expensive, we should use `map_or_else`, but it's not so this is perfectly fine.